### PR TITLE
Improve font lock and indent

### PR DESCRIPTION
**Pull request by dmilstein**
Hello,

First off, thanks very much for creating this emacs mode! I've been exploring TLA+ and Pluscal lately, and being able to start with a functioning emacs mode, instead of having to use the Toolbox, has been an absolute godsend.

Second: I've been doing a bunch of hacking on the mode -- adding font-lock coverage for more of the pcal/tla+ features, fixing a bunch of indentation issues, and adding some new indentation support (e.g. to indent inside of parens or braces).

I'm submitting a bunch of (mostly smallish) commits as one big PR. If you'd prefer some other style of collaboration, let me know what's useful to you. I tried to make the individual commit messages fairly clear as to what each was doing (and added a decent chunk of ert tests to make it even clearer).

### CODE REVIEW

File: tla-pcal-mode.el [hunk 1]
```diff
@@ -280,18 +282,23 @@ nil if the syntax isn't recognized for indentation."
 		 (8  "        skip;")
 		 (4  "    end while;")
 		 (0  "end"))))
-    (dotimes (n 100)
+    (dotimes (n (length lines))
```
Comment (mrc) [inline tla-pcal-mode.el]: Nice change, thanks!

File: test/tla-pcal-mode-test.el [hunk 2]
```diff
@@ -0,0 +1,78 @@
+;;; tla-pcal-mode-test.el --- testing for TLA and PlusCal modes -*- lexical-binding: t; -*-
```
Comment (mrc) [inline test/tla-pcal-mode-test.el:1]: Thanks!

**Review by mrc — CHANGES_REQUESTED** [review]
Hi, this is fantastic, thank you!

I see you set the tab width, but I prefer spaces for indentation, so do you mind running `untabify`?

**Comment by mrc** [comment]
To answer your comments: I really do like this style, reviewing each commit was helpful for me, and I appreciate your commit messages which show each change is thoughtful. The only sticking point I have is the tabs, do you mind removing them?

About the tests: I've been thinking it would be better if the tests were in example files which could be read in and the indent width inferred for each line, rather than explicit. It would make it easier to drop in new and more complex files (I was working through "Practical TLA+" and making the tests more complex as I got through more of the book.) The explicit line-by-line tests came from an early approach I took when I had only a few simple test cases, but as it grew it became a bit unwieldily.

Seeing as you're contributing to this and adding tests, I figure you might have some good ideas here, so let me know what you think.

**Comment by dmilstein** [comment]
Got rid of the tabs (I am strongly with you, and am very happy to move to spaces).

Example files for tests: I love this idea. I'm hoping I'll have time to do a bit more hacking, and will see if I can implement this as/when I get into that (to be clear: I am leveling up on elisp in order to do all this, so I'm not, like, an expert ;-)

**Comment by mrc** [comment]
Thank you!!

**Pull request merged**